src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/service/VisitService.kt

```diff
@@ -73,53 +83,131 @@ class VisitService(
   }
 
   fun cancelVisit(offenderNo: String, visitId: Long, visitDto: CancelVisitRequest) {
+    val today = LocalDate.now()
+    val visit = visitRepository.findById(visitId).orElseThrow(PrisonerNotFoundException(offenderNo))
```

Comment:
is it worth having a VisitNotFoundException, it might be confusing if this scenario ever happens and we are trying to work out why its can't find the prisoner?

Reply:
will do